firetweet/src/main/java/org/getlantern/firetweet/activity/support/UpdaterActivity.java

```diff
@@ -0,0 +1,175 @@
+
+        Typeface font = Typeface.createFromAsset(getAssets(), "fonts/ProximaNova-Semibold.ttf");
+        updateAvailableText = (TextView)findViewById(R.id.update_available);
+        updateAvailableText.setTextColor(Color.parseColor("black"));
+        updateAvailableText.setTypeface(font);
+
+        Button btn=(Button) findViewById(R.id.not_now);
+        btn.setTypeface(font);
+        btn.setOnClickListener(new OnClickListener() {
+
+            @Override
+            public void onClick(View v) {
+                finish();
+            }
+        });
+
+        btn = (Button)findViewById(R.id.install_update);
+        btn.setTypeface(font);
+        btn.setOnClickListener(new OnClickListener() {
+
+            @Override
+            public void onClick(View v) {
+                runUpdater();
+            }
+        });
+    }
+
+    private void runUpdater() {
+        updateIcon.setVisibility(View.GONE);
+        spinner.setVisibility(View.VISIBLE);
+
+        String[] updaterParams = {APK_URL};
+        mUpdaterTask = new UpdaterTask(this);
+        mUpdaterTask.execute(updaterParams);
+    }
+
+    public void dismissActivity() {
+
+        finish();
+    }
+
+
+    public void dismissActivity(View v) {
+        finish();
+    }
+
+
+
+    @Override
+    public void finish() {
+        super.finish();
+    }
+
+    static class UpdaterTask extends AsyncTask<String, String, String> {
+
+        private final UpdaterActivity mActivity;
+        private final Context context;
+
+        private static final String APK_PATH = "/sdcard/FireTweet.apk";
+
+        UpdaterTask(final UpdaterActivity activity) {
+            mActivity = activity;
+            context = mActivity.getApplicationContext();
+        }
+
+        @Override
+        protected String doInBackground(String... sUrl) {
+            String path = APK_PATH;
+            try {
+                Log.d(LOG_TAG, "Attempting to download new APK from " + sUrl[0]);
+                URL url = new URL(sUrl[0]);
+                URLConnection connection = url.openConnection();
+                connection.connect();
+
+                int fileLength = connection.getContentLength();
+
+                // download the file
+                InputStream input = new BufferedInputStream(url.openStream());
+                OutputStream output = new FileOutputStream(path);
+
+                byte data[] = new byte[1024];
+                long total = 0;
+                int count;
+                while ((count = input.read(data)) != -1) {
+                    total += count;
+                    int progress = (int) (total * 100 / fileLength);
+                    publishProgress(Integer.toString(progress));
+                    output.write(data, 0, count);
+                }
+
+                output.flush();
+                output.close();
+                input.close();
+            } catch (Exception e) {
+                Log.e(LOG_TAG, "Error installing new APK..");
```

Comment:
Guess we can return null path and check it in `onPostExecute`. Furthermore, as `path` is always `APK_PATH`, we can return `Boolean` instead.